bot/cogs/search.py

```diff
@@ -201,68 +201,144 @@ async def anime(self, ctx: Context, *, query: str) -> None:
+                        await ctx.send(template)
+
+    @command()
+    async def weather(self, ctx: Context, *, city: str = None) -> None:
+        """Sends current weather in the given city name
+
+        eg. weather london"""
+        url_formatted_city = city.replace(" ", "-")
+        async with aiohttp.ClientSession() as session:
+            try:
+                weather_lookup_url = f"https://api.openweathermap.org/data/2.5/weather?q={url_formatted_city}&appid={config.WEATHER_API_KEY}"
+                async with session.get(weather_lookup_url) as resp:
+                    data = await resp.json()
+            except Exception:
+                await ctx.send("You didn't provide a city name")
+                return
+
+        if data["cod"] == "401":
+            await ctx.send("Invalid API key")
+            return
+        if data["cod"] == "404":
+            await ctx.send("Invalid city name")
+            return
+
+        weather_embed = Embed(
+            title=f"Current Weather in {city.capitalize()}",
+            color=Color.blue()
+        )
+        longtitude = data["coord"]["lon"]
+        lattitude = data["coord"]["lat"]
+        weather_embed.add_field(
+            name="Coordinates",
+            value=f"**❯❯ Longtitude: **`{longtitude}`\n**❯❯ Latittude: **`{lattitude}`"
+        )
+        actual_temp = round(data["main"]["temp"] / 10, 1)
+        feels_like = round(data["main"]["feels_like"] / 10, 1)
+        weather_embed.add_field(
+            name="Temperature",
+            value=f"**❯❯ Temperature: **`{actual_temp}°C`\n**❯❯ Feels Like: **`{feels_like}°C`"
+        )
+        wind_speed = data["wind"]["speed"]
+        wind_direction = data["wind"]["deg"]
+        weather_embed.add_field(
+            name="Wind",
+            value=f"**❯❯ Speed: **`{wind_speed}km/h`\n**❯❯ Direction: **`{wind_direction}°`",
+        )
+        visibility = round(data["visibility"] / 1000, 2)
+        humidity = data["main"]["humidity"]
+        weather_description = data["weather"][0]["description"]
+        weather_embed.add_field(
+            name="Miscellaneous",
+            value=f"**❯❯ Humidity: **`{humidity}%`\n**❯❯ Visibility: **`{visibility}km`\n**❯❯ Weather Summary: **`{weather_description}`",
+        )
+
+        weather_icons = {
+            "wind": "https://cdn.discordapp.com/attachments/728569086174298112/735550169222873118/windy.png",
+            "rain": "https://cdn.discordapp.com/attachments/728569086174298112/735550164458274947/raining.png",
+            "sun": "https://cdn.discordapp.com/attachments/728569086174298112/735550167859593306/sunny.png",
+            "cloud": "https://cdn.discordapp.com/attachments/728569086174298112/735550159781494865/cloudy.png",
+            "partly": "https://cdn.discordapp.com/attachments/728569086174298112/735550162721701979/partly.png",
+            "snow": "https://cdn.discordapp.com/attachments/728569086174298112/735550166563684474/snowy.png"
+        }
```

Comment:
This should be a part of [`config.py`](https://github.com/The-Codin-Hole/HotWired-Bot/blob/master/bot/config.py)